src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -96,11 +97,12 @@ private function resolveArguments(ContainerBuilder $container, array $arguments,
      */
     private function resolveDefinition(ContainerBuilder $container, DefinitionDecorator $definition)
     {
-        if (!$container->hasDefinition($parent = $definition->getParent())) {
```

Comment:
another solution is to use ``$container->has()`` to validate the direct parent, keeping the existing error message as is, and then simply using ``findDefinition`` below (and letting it throw its exception if an alias cannot be resolved, as the message will be clear in this case)